src/repositoryModals.jsx

```diff
@@ -0,0 +1,451 @@
+            </Form>
+        </Modal>
+    );
+};
+
+RemoveRepositoryModal.propTypes = {
+    origin: PropTypes.object.isRequired,
+    availableRemotes: PropTypes.array.isRequired,
+    refreshRemotes: PropTypes.func.isRequired,
+};
+
+export const AddRepositoryModal = ({ refreshRemotes }) => {
+    const Dialogs = useDialogs();
+    const [newRepoName, setNewRepoName] = useState("");
+    const [newRepoURL, setNewRepoURL] = useState("");
+    const [newRepoTrusted, setNewRepoTrusted] = useState(false);
+
+    const [hasValidation, setHasValidation] = useState(false);
+    const [addNewRepoError, setAddNewRepoError] = useState(undefined);
+
+    const onAddRemote = () => {
+        if (!(newRepoURL.trim().length && newRepoName.trim().length)) {
+            setHasValidation(true);
+            return;
+        }
+        return remotes.addRemote(newRepoName, newRepoURL, newRepoTrusted)
+                .then(() => refreshRemotes())
+                .then(Dialogs.close,
+                      ex => setAddNewRepoError(ex.message));
+    };
+
+    const actions = [
+        <Button key="add-repo"
+            variant="primary"
+            onClick={() => onAddRemote()}>
+            {_("Add repository")}
+        </Button>,
+        <Button key="cancel" variant="link" onClick={Dialogs.close}>
+            {_("Cancel")}
+        </Button>
+    ];
+
+    return (
+        <Modal isOpen
+            id="add-repository-modal"
+            title={_("Add new repository")}
+            position="top"
+            variant="small"
+            onClose={Dialogs.close}
+            actions={actions}
+        >
+            <Form isHorizontal>
+                {addNewRepoError && <Alert variant="danger" isInline title={addNewRepoError} />}
+                <FormGroup label={_("Name")}
+                    fieldId="new-remote-name"
+                    isRequired
+                >
+                    <TextInput id="new-remote-name"
+                        value={newRepoName}
+                        isRequired
+                        type="text"
+                        onChange={(_ev, name) => setNewRepoName(name)}
+                    />
+                    <FormHelper fieldId="new-remote-name"
+                        helperTextInvalid={(hasValidation && !newRepoName.trim().length) && _("Please provide a valid name")}
+                    />
+                </FormGroup>
+                <FormGroup label={_("URL")}
+                    fieldId="new-remote-url"
+                    isRequired
+                >
+                    <TextInput id="new-remote-url"
+                        value={newRepoURL}
+                        isRequired
+                        type="text"
+                        onChange={(_ev, url) => setNewRepoURL(url)}
+                    />
+                    <FormHelper fieldId="new-remote-url"
+                        helperTextInvalid={(hasValidation && !newRepoURL.trim().length) && _("Please provide a valid URL")}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231128-111939-af484d30-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)